src/System.Private.CoreLib/shared/System/Text/Rune.cs

```diff
@@ -333,11 +335,14 @@ private static int ReadRuneFromString(string input, int index)
         /// <summary>
         /// Returns a <see cref="string"/> representation of this <see cref="Rune"/> instance.
         /// </summary>
-        public override string ToString()
-        {
-            Span<char> chars = stackalloc char[2]; // worst case
-            return new string(chars.Slice(0, EncodeToUtf16(chars)));
-        }
+        public override string ToString() =>
+            IsBmp ? char.ToString((char)_value) :
+            string.Create(2, _value, (span, value) =>
+            {
+                ref char dest = ref MemoryMarshal.GetReference(span);
+                UnicodeUtility.GetUtf16SurrogatesFromSupplementaryPlaneScalar(
```

Comment:
I would make this either all nice and safe at the cost of some performance. Or if we want to go with the max performance, I would add this method to `string`:

```
        internal static string CreateFromChar(char c1, char c2)
        {
            string result = FastAllocateString(2);
            result._firstChar = c1;
            Unsafe.Add(ref result._firstChar, 1) = c2;
            return result;
        }
```

This would be produce faster and smaller code than this.

Reply:
Done.